src/shared/store/albumSlice.ts

```diff
@@ -0,0 +1,90 @@
+import {createAsyncThunk, createSlice} from '@reduxjs/toolkit';
+
+import API from '../../core/services/API';
+import {IAlbum} from '../interfaces';
+import {IAlbumState} from '../interfaces/StateSlices';
+import {isFulfilled, isFulfilledAction, isPending, isPendingAction, isRejected, isRejectedAction} from './helpers';
+
+
+export const addAlbum = createAsyncThunk(
+  'photo/addAlbum',
+  async function (album: IAlbum, thunkAPI) {
+    try {
+      return await API.post('/api/albums', album) as IAlbum;
+    } catch (error) {
+      return thunkAPI.rejectWithValue('Error');
+    }
+  }
+);
+
+export const updateAlbum = createAsyncThunk(
+  'photo/updateAlbum',
+  async function (album: Required<IAlbum>, thunkAPI) {
+    try {
+      return await API.patch('/api/albums', album) as IAlbum;
+    } catch (error) {
+      return thunkAPI.rejectWithValue('Error');
+    }
+  }
+);
+
+export const getAlbum = createAsyncThunk(
+  'photo/getAlbum',
+  async function (id: number[], thunkAPI) {
+    try {
+      return await API.get(`/api/albums${id.length === 0 ? '' : `?ids=${id.join()}`}`) as IAlbum[] | IAlbum;
```

Comment:
This wasn't changed. You need to specify when it is `IAlbum[]` and when it's `Album`
Same for photos